kernelci/cli/event.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+"""Tool to interact with the Pub/Sub interface"""
+
+import sys
+import json
+
+import click
+
+import kernelci.api
+import kernelci.api.helper
+import kernelci.config
+from . import Args, kci
+
+
+@kci.group(name='event')
+def kci_event():
+    """Interact with Pub/Sub interface"""
+
+
+@kci_event.command(secrets=True)
+@click.argument('channel')
+@Args.config
+@Args.api
+def subscribe(config, api, channel, secrets):
+    """Subscribe to a Pub/Sub channel"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    sub_id = api.subscribe(channel)
+    click.echo(sub_id)
+
+
+@kci_event.command(secrets=True)
+@click.argument('sub_id')
+@Args.config
+@Args.api
+def unsubscribe(config, api, sub_id, secrets):
+    """Unsubscribe from a Pub/Sub channel"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    api.unsubscribe(sub_id)
+
+
+@kci_event.command(secrets=True)
+@click.option('--event-type',
+              help="CloudEvent type, api.kernelci.org by default")
+@click.option('--source',
+              help="CloudEvent source, https://api.kernelci.org/ by default")
+@click.option('--json-data', help="Parse input data as JSON", is_flag=True)
+@Args.config
+@Args.api
+@click.argument('channel')
+def send(config, api, event_type, source,  # pylint: disable=too-many-arguments
+         json_data, channel, secrets):
+    """Read some data on stdin and send it as an event on a channel"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    data = sys.stdin.read()
+    if json_data:
+        data = json.loads(data)
+    attributes = {}
+    if event_type:
+        attributes['type'] = event_type
+    if source:
+        attributes['source'] = source
+    attributes['data'] = data
+    api.send_event(channel, attributes)
+
+
+@kci_event.command(secrets=True)
+@click.argument('sub_id')
+@Args.config
+@Args.api
+@Args.indent
+def receive(config, api, indent, sub_id, secrets):
```

Comment:
How about a timeout?  Maybe the API doesn't handle that yet actually.

Reply:
`PubSub.listen` handler has set static value for timeout i.e. 1 sec. I also used the same value for blocking pop with Redis List.

Reply:
Right, so there's no support for it yet.  That's fine, we can add it later.